catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -164,6 +170,8 @@ private ResultIterable(QueryFunction queryFunction, QueryRequest queryRequest,
 
         private boolean finished = false;
 
+        private final Set<String> foundIds = new HashSet<>(2048);
```

Comment:
❓ Rational?
Many queries might just be 20 items. or 250.  2048 default set size seems a little extreme since it will be very difficult to gauge how much it will actually need

Reply:
If the resultset _is_ large, the last thing we want to do is thrash and repeatedly perform mem copies. If the resultset is small, this is going to be very short-lived. Also, it's not realIy all that much memory being allocated up front. I weighed the options and leaned toward the more memory-intensive solution.

I could be persuaded to change the default size.